nssp/delphi_nssp/patch.py

```diff
@@ -0,0 +1,139 @@
+"""
+This module is used for patching data in the delphi_nssp package.
+
+To use this module, you need to turn on the custom_run flag
+and specify the range of issue dates in params.json, like so:
+
+{
+  "common": {
+    "custom_run": true,
+    ...
+  },
+  "validation": {
+    ...
+  },
+  "patch": {
+    "patch_dir": "/Users/minhkhuele/Desktop/delphi/covidcast-indicators/nssp/AprilPatch",
+    "start_issue": "2024-04-20",
+    "end_issue": "2024-04-21",
+    "source_dir": "/Users/minhkhuele/Desktop/delphi/covidcast-indicators/nssp/source_data"
+  }
+}
+
+It will generate data for that range of issue dates, and store them in batch issue format:
+[name-of-patch]/issue_[issue-date]/nssp/actual_data_file.csv
+"""
+
+import sys
+from datetime import datetime, timedelta
+from os import makedirs, path
+
+from delphi_utils import get_structured_logger, read_params
+from epiweeks import Week
+
+from .run import run_module
+
+
+def good_patch_config(params, logger):
```

Comment:
I wonder if we can bake this in the read_params() in delphi_utils.... @nmdefries thoughts?
pydantic doesn't a full pledged support for json schema validation, but seeing how the read_params has no validation what so ever, we could look into it sometime in the future.

Reply:
I agree. I wrote this method and half way through was like hmm this should probably be generalized since it may be applicable to other indicators too.

Reply:
yeahh I'm like oh I also would like that for mine  👀 and also saw that the current read params isn't doing any validations at all

Reply:
I mentioned it the #2002 and I think for right now it's better to remove this and deal with this in another issue